tck/methods/token.go

```diff
@@ -0,0 +1,264 @@
+		if err != nil {
+			return nil, err
+		}
+		transaction.SetInitialSupply(uint64(initialSupply))
+	}
+	if params.TreasuryAccountId != nil {
+		accountID, err := hiero.AccountIDFromString(*params.TreasuryAccountId)
+		if err != nil {
+			return nil, err
+		}
+		transaction.SetTreasuryAccountID(accountID)
+	}
+	if params.FreezeDefault != nil {
+		transaction.SetFreezeDefault(*params.FreezeDefault)
+	}
+	if params.ExpirationTime != nil {
+		expirationTime, err := strconv.ParseInt(*params.ExpirationTime, 10, 64)
+		if err != nil {
+			return nil, err
+		}
+		transaction.SetExpirationTime(time.Unix(expirationTime, 0))
+	}
+	if params.AutoRenewAccountId != nil {
+		autoRenewAccountId, err := hiero.AccountIDFromString(*params.AutoRenewAccountId)
+		if err != nil {
+			return nil, err
+		}
+		transaction.SetAutoRenewAccount(autoRenewAccountId)
+	}
+	if params.AutoRenewPeriod != nil {
+		autoRenewPeriodSeconds, err := strconv.ParseInt(*params.AutoRenewPeriod, 10, 64)
+		if err != nil {
+			return nil, err
+		}
+		transaction.SetAutoRenewPeriod(time.Duration(autoRenewPeriodSeconds) * time.Second)
+	}
+
+	if params.Metadata != nil {
+		transaction.SetTokenMetadata([]byte(*params.Metadata))
+	}
+
+	if params.CommonTransactionParams != nil {
+		params.CommonTransactionParams.FillOutTransaction(transaction, t.sdkService.Client)
+	}
+
+	if params.CustomFees != nil {
+		var customFeeList []hiero.Fee
+		for _, customFee := range *params.CustomFees {
+			// Handle Fixed Fee
+			if customFee.FixedFee != nil {
+				fee := hiero.NewCustomFixedFee()
+				fee.SetAmount(ParseIntFromOptional(&customFee.FixedFee.Amount))
+				feeCollector, err := hiero.AccountIDFromString(customFee.FeeCollectorAccountId)
+				if err != nil {
+					return nil, err
+				}
+				fee.SetFeeCollectorAccountID(feeCollector)
+				fee.SetAllCollectorsAreExempt(*customFee.FeeCollectorsExempt)
+				if customFee.FixedFee.DenominatingTokenId != nil {
+					tokenId, err := hiero.TokenIDFromString(*customFee.FixedFee.DenominatingTokenId)
+					if err != nil {
+						return nil, err
+					}
+					fee.SetDenominatingTokenID(tokenId)
+				}
+				customFeeList = append(customFeeList, fee)
+			}
+
+			// Handle Fractional Fee
+			if customFee.FractionalFee != nil {
+				fee := hiero.NewCustomFractionalFee()
+				fee.SetNumerator(ParseIntFromOptional(&customFee.FractionalFee.Numerator))
+				fee.SetDenominator(ParseIntFromOptional(&customFee.FractionalFee.Denominator))
+				fee.SetMin(ParseIntFromOptional(&customFee.FractionalFee.MinimumAmount))
+				fee.SetMax(ParseIntFromOptional(&customFee.FractionalFee.MaximumAmount))
+				feeCollector, err := hiero.AccountIDFromString(customFee.FeeCollectorAccountId)
+				if err != nil {
+					return nil, err
+				}
+				fee.SetFeeCollectorAccountID(feeCollector)
+				fee.SetAllCollectorsAreExempt(*customFee.FeeCollectorsExempt)
+				customFeeList = append(customFeeList, fee)
+			}
+
+			// Handle Royalty Fee
```

Comment:
If we expect to have `customFee`s in other transactions like `TokenFeeScheduleUpdateTransaction` or potentially other places maybe we can reuse the marshalling logic.

Reply:
nice catch, done